lib/grpfp.gi

```diff
@@ -955,6 +955,11 @@ local iso,hom,u;
   if HasIsAbelian(G) and IsAbelian(G) then
     return TrivialSubgroup(G);
   elif Size(Image(hom))=infinity then
+    # test a special case -- one generator
+    if Length(GeneratorsOfGroup(G))=1 then
+      SetIsAbelian(G,true);
+      return TrivialSubgroup(G);
+    fi;
```

Comment:
So this seems to be a separate improvement for `DerivedSubgroup` on fp subgroups of rank 1... But why is it needed? Because we removed the immediate method for "size of generating set <= 1 implies abelian" ? 
This raises two questions:

1. Where is the place we create a group w/o taking care of that? can we do something about it (i.e. fix the issue at its root, not the symptom)
2. Do we really need to *remove* (or replace by normal ones) those immediate methods? After all, if e.g. `IsFinite` is set, then no immediate method for `IsFinite` is run. So leaving those immediate methods in should not cause a speed penalty anymore -- or does it? I should run some tests....